data/network/src/commonMain/kotlin/social/androiddev/common/network/util/RunCatching.kt

```diff
@@ -0,0 +1,46 @@
+package social.androiddev.common.network.util
```

Comment:
TODO: this file should get moved to a new standalone kotlin-utils module that any other module can depend on.

Reply:
Rec something like `core-util`? Not sure how important the language name is here.

Reply:
I like it

Reply:
I'm not a big fan of these extra modules. In big projects you'll end up in a really big module every module depends on.
With high-level functions and kotlin extensions these function can be reduced to a minimum anyway.

Reply:
Ok, I suppose they could just be moved to `domain` since most likely a lot of the other modules will depend on that anyway. I don't feel strongly over this but I do think over time we will find helpful functions/extensions/utils that would/could be used across the whole project.

Reply:
Yes, we might also have the domain layer splitted soon.
domain:authentication
domain:timeline
...